src/m_menu.c

```diff
@@ -2538,10 +2543,7 @@ static void M_DrawInstructions()
         }
         else if (flags & S_RESET)
         {
-            if (menu_input == pad_mode)
-                s = "Press PadA to reset to defaults";
-            else
-                s = "Press Enter key to reset to defaults";
+            s = "Reset submenu to defaults";
```

Comment:
I need help with wording here and in other help messages. What if we remove "Press", e.g. "Left Button to change, Del to clear"? Also, do we need so many help messages?

Reply:
"Restore Defaults" is good. We don't have enough resolution to provide useful descriptions of what menu items do. Should we remove all other help text except that one? I don't think they're needed.

Another solution: If any help text is shown at all, it only needs to be something like this, shown at all times:

When using mouse/keyboard:
```
[ESC] Back     [R] Restore Defaults
```

When using a gamepad (have to query SDL to get correct Xbox/PS button names):
```
[B] Back       [Y] Restore Defaults
```

The skull icon is moved back to the upper left and the reset icon is removed completely.

Reply:
Even the back text isn't really needed. Just "Restore Defaults".

Reply:
Yet another idea: A dedicated menu item called "Restore Defaults" on the "General > Misc" tab and nowhere else that completely restores all defaults for everything. The bottom row is empty (except the skull if you prefer).

Reply:
> When using a gamepad (have to query SDL to get correct Xbox/PS button names)

We use the names "PadA", "PadB" in the key bindings. We don't have space or icons for "Circle"/"Triangle" etc. Last time I checked this, SDL2 didn't have PS buttons names, but controller support has been updated.

> Should we remove all other help text except that one? I don't think they're needed.

I think we should leave some help messages for the keyboard/gamepad, it's not obvious that there's a "select mode" in the menu.

> When using mouse/keyboard:
> 
> ```
> [ESC] Back     [R] Restore Defaults
> ```

I like the idea.

Reply:
It's not so simple I guess, because I forgot about `[BKSP] Previous Menu` or something similar.

Reply:
I tweaked the messages a bit: https://github.com/fabiangreffrath/woof/pull/1402/commits/c642a8433cba991317f76f69a50bf125905e4d33

Maybe we should postpone it for now. @fabiangreffrath I think everything is done.

Reply:
The reset button doesn*t have a string when selected by keyboard.

Reply:
> The reset button doesn*t have a string when selected by keyboard.

I force-pushed the fix.

Reply:
It's really great now, thank you! One last thing, what if we increase `KB_X` just a little bit?

Reply:
Quick tweaks:

```diff
diff --git a/src/m_menu.c b/src/m_menu.c
index 4bf1cb7..b163f7e 100644
--- a/src/m_menu.c
+++ b/src/m_menu.c
@@ -2397,11 +2397,11 @@ static void M_DrawScreenItems(setup_menu_t* src)
     }
     else if (warning_about_changes & S_PRGWARN)
     {
-      strcpy(menu_buffer, "Warning: Program must be restarted to see changes");
+      strcpy(menu_buffer, "Warning: Restart to see changes");
     }
     else
     {
-      strcpy(menu_buffer, "Warning: Changes are pending until next game");
+      strcpy(menu_buffer, "Warning: Changes apply to next game");
     }
 
     x_warn = SCREENWIDTH/2 - M_GetPixelWidth(menu_buffer)/2;
@@ -2455,7 +2455,7 @@ static void M_DrawDefVerify()
 
   if (whichSkull) // blink the text
     {
-      strcpy(menu_buffer,"Reset to defaults? (Y or N)");
+      strcpy(menu_buffer,"Restore defaults? (Y or N)");
       M_DrawMenuString(VERIFYBOXXORG+8,VERIFYBOXYORG+8,CR_RED);
     }
 }
@@ -2501,7 +2501,7 @@ static void M_DrawInstructions()
     {
         if (!(setup_gather | print_warning_about_changes))
         {
-            strcpy(menu_buffer, "Current actual setting differs from the default.");
+            strcpy(menu_buffer, "Current value differs from default");
             M_DrawMenuString(4, M_Y_WARN, CR_RED);
         }
```

Reply:
Doom brackets `[` and `]` don't look great, plus small gamepad tweak:

```diff
diff --git a/src/m_menu.c b/src/m_menu.c
index 4bf1cb7..c03672d 100644
--- a/src/m_menu.c
+++ b/src/m_menu.c
@@ -2525,13 +2525,16 @@ static void M_DrawInstructions()
         else if (flags & S_YESNO)
         {
             if (menu_input == pad_mode)
-                s = "[ PadA ] toggle";
+                s = "'PadA' to toggle";
             else
-                s = "[ Enter ] toggle, [ Esc ] cancel";
+                s = "'Enter' to toggle, 'Esc' to cancel";
         }
         else if (flags & (S_CHOICE|S_CRITEM|S_THERMO))
         {
-            s = "[ Left ] [ Right ] choose, [ Esc ] cancel";
+            if (menu_input == pad_mode)
+                s = "'Left/Right' to choose, 'PadB' to cancel";
+            else
+                s = "'Left/Right' to choose, 'Esc' to cancel";
         }
         else if (flags & S_NUM)
         {
@@ -2553,14 +2556,14 @@ static void M_DrawInstructions()
             switch (menu_input)
             {
                 case mouse_mode:
-                    s = "[ Del ] clear";
+                    s = "'Del' to clear";
                     break;
                 case pad_mode:
-                    s = "[ PadA ] change, [ PadY ] clear";
+                    s = "'PadA' to change, 'PadY' to clear";
                     break;
                 default:
                 case key_mode:
-                    s = "[ Enter ] change, [ Del ] clear";
+                    s = "'Enter' to change, 'Del' to clear";
                     break;
             }
         }
@@ -2573,10 +2576,10 @@ static void M_DrawInstructions()
             switch (menu_input)
             {
                 case pad_mode:
-                    s = "[ PadA ] change, [ PadB ] back";
+                    s = "'PadA' to change, 'PadB' to return";
                     break;
                 case key_mode:
-                    s = "[ Enter ] change";
+                    s = "'Enter' to change";
                     break;
                 default:
                     break;
```

Reply:
> One last thing, what if we increase `KB_X` just a little bit?

Sure, but why? Currently the user can bind a maximum of 4 inputs to an element, but after updating `m_input.c` in this PR we can allow any number. Maybe we should increase the limit to 6 for example.

Reply:
> Doom brackets `[` and `]` don't look great

I think `'` looks worse. Maybe we should do it in color, like the original help screen:
![woof0002](https://github.com/fabiangreffrath/woof/assets/5077629/4bb21ca1-455d-4bcd-9657-0dc647857795)

Reply:
Good idea! Experiment a bit.

Reply:
Color is even better, takes less space and it's unambigious what is the key and what is its result.

Reply:
> Sure, but why? Currently the user can bind a maximum of 4 inputs to an element, but after updating `m_input.c` in this PR we can allow any number. Maybe we should increase the limit to 6 for example.

Because I found that some of the key strings would come a bit close to the left screen edge, especially for PWADs with wider fonts.

Reply:
Fixed here: https://github.com/fabiangreffrath/woof/pull/1402/commits/487a62822cbcf181490e27d07768d55f6bd0ff9b

Reply:
This seems reasonable:
- `Enter` and `Del` are regular red.
- `to change` and `to clear` are slightly darker red: `colormaps[0][256*11]`.

![input](https://github.com/fabiangreffrath/woof/assets/56656010/007a5a2b-9b5b-470a-86f4-0a0422c2c741)

Reply:
Maybe the last nitpick, but the underline under the category name is much brighter than the font. Should we probably colorize some mid-gray instead of the lightest available color?

Reply:
Slightly darker inactive tabs (just an example hack using `hsv.z *= 0.8;`, not sure how it's supposed to be done):

![tabs](https://github.com/fabiangreffrath/woof/assets/56656010/08835e9a-b619-47ea-8e93-8f505d5af5cc)

The underline is also slightly darker.

Reply:
Btw, the most common color blindness is red-green.

Reply:
> * `Enter` and `Del` are regular red.
> * `to change` and `to clear` are slightly darker red: `colormaps[0][256*11]`.

I find this a bit dull to be honest.

Reply:
> > * `Enter` and `Del` are regular red.
> > * `to change` and `to clear` are slightly darker red: `colormaps[0][256*11]`.
> 
> I find this a bit dull to be honest.

Change `11` to preference, but the help text should be somewhat dull to not draw any attention.

Reply:
> Change `11` to preference, but the help text should be somewhat dull to not draw any attention.

Hm, I still prefer brick, i.e. *desaturated* red over just *darker* red. But I think we shouldn't go to deep into the rabbit hole of personal preference. :wink:

Reply:
So what's the consensus? I like the dull version, I always thought Boom was too colorful. Also, some PWADs modify CR* lumps, essentially turning off the colors, BTSX 2 and 3 in particular:
![woof0002](https://github.com/fabiangreffrath/woof/assets/5077629/abba89a9-d94e-4cfb-a812-cdf2d606257a)

Everyone's favorite Ancient Aliens has an interesting font:
![woof0003](https://github.com/fabiangreffrath/woof/assets/5077629/ddebad16-ab40-4880-9bfb-68684cf14fbf)
Should we do something about it?

Reply:
> So what's the consensus? I like the dull version, I always thought Boom was too colorful. 

What if we draw the entire line in the shade tone?

Reply:
> What if we draw the entire line in the shade tone?

Do you mean help line? Yes, I like it. Should we mark keys/buttons with a slightly lighter shade?

If you ask me, I would remove most colors and use bright/shade instead.

Reply:
I would leave the entire help line in the same darker shade (remember, before this refactoring it was entirely golden).

I am also against adding even more colors. However, I find that changes in shading between words are just as disruptive. As @ceski-1 pointed out, the help line is supposed to be boring and unobtrusive.

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/1402/commits/39e10572de553a7aa1d812e4f0991b4bdc095214

Reply:
> Maybe the last nitpick, but the underline under the category name is much brighter than the font. Should we probably colorize some mid-gray instead of the lightest available color?

Maybe like this?

```patch
--- a/src/m_menu.c
+++ b/src/m_menu.c
@@ -2083,7 +2083,7 @@ static void M_DrawTabs(void)
         if (i == mult_screens_index)
         {
             V_FillRect(x + video.deltaw, rect->y + M_SPC, rect->w, 1,
-                       cr_gold[v_lightest_color]);
+                       cr_shaded[cr_gold[v_lightest_color]]);
         }
 
         rect->x = x;
```

Reply:
> OK, but I don't think it changes much. Sometimes this line is just white (Dimension of the Boomed):

Hm, that doesn't even look bad.

> Maybe we shouldn't use `v_lightest_color`?

Or `v_midgray_color = I_GetPaletteIndex(playpal, 0x80, 0x80, 0x80);`?

Reply:
Yes, or `cr_gold[cr_shaded[v_lightest_color]]`, I don't know.